pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -197,9 +200,151 @@ func (r *ReconcileUnifiedPushServer) Reconcile(request reconcile.Request) (recon
+				return reconcile.Result{}, err
+			}
+
+			return reconcile.Result{RequeueAfter: time.Second * 1}, nil
+		} else if err != nil {
+			return reconcile.Result{}, err
+		} /*User exists and is ready*/
+
+		/*User exists and is ready*/
+		//check that addresses exist
+		//queues
+		queues := []string{"APNsPushMessageQueue", "APNsTokenBatchQueue", "GCMPushMessageQueue", "GCMTokenBatchQueue", "WNSPushMessageQueue", "WNSTokenBatchQueue", "MetricsQueue", "TriggerMetricCollectionQueue", "TriggerVariantMetricCollectionQueue", "BatchLoadedQueue", "AllBatchesLoadedQueue", "FreeServiceSlotQueue"}
+		requeueCreate := false
+		for _, address := range queues {
+			queue := newQueue(instance, address)
+			foundQueue := &enmassev1beta.Address{}
+			// Set UnifiedPushServer instance as the owner and controller
+			if err := controllerutil.SetControllerReference(instance, queue, r.scheme); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			err = r.client.Get(context.TODO(), types.NamespacedName{Name: queue.Name, Namespace: queue.Namespace}, foundQueue)
+			if err != nil && errors.IsNotFound(err) {
+				reqLogger.Info("Creating a new Queue", "Queue.Namespace", queue.Namespace, "Queue.Name", queue.Name)
+				err = r.client.Create(context.TODO(), queue)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+				requeueCreate = true
+			} else if err != nil {
+				reqLogger.Info("Queue Error")
+				return reconcile.Result{}, err
+			} else if !foundQueue.Status.IsReady {
+				reqLogger.Info("Queue Not ready", "Queue.Name", foundQueue.Name)
+				requeueCreate = true
+			}
+		}
+
+		if requeueCreate {
+			return reconcile.Result{RequeueAfter: time.Second * 5}, nil
+		}
+
+		reqLogger.Info("Found All queues  for UPS")
+
+		//topics
+		topics := []string{"MetricsProcessingStartedTopic", "topic/APNSClient"}
+		for _, address := range topics {
+			topic := newTopic(instance, address)
+			foundTopic := &enmassev1beta.Address{}
+			// Set UnifiedPushServer instance as the owner and controller
+			if err := controllerutil.SetControllerReference(instance, topic, r.scheme); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			err = r.client.Get(context.TODO(), types.NamespacedName{Name: topic.Name, Namespace: topic.Namespace}, foundTopic)
+			if err != nil && errors.IsNotFound(err) {
+				reqLogger.Info("Creating a new Topic", "Topic.Namespace", topic.Namespace, "Topic.Name", topic.Name)
+				err = r.client.Create(context.TODO(), topic)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+				requeueCreate = true
+			} else if err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+
+		if requeueCreate {
+			return reconcile.Result{RequeueAfter: time.Second * 5}, nil
+		}
+
+		reqLogger.Info("Found All queues and topics for UPS")
+
+	}
+	//If AMQ is used, it is ready
+
 	//#region Postgres PVC
 	persistentVolumeClaim, err := newPostgresqlPersistentVolumeClaim(instance)
 	if err != nil {
+		reqLogger.Info("Error making PVC")
```

Comment:
This seems unrelated to everything else. Have you had issues with it (and the logs that would be caused by returning with the error in the line below)? The log message, without displaying anything about the error, seems incomplete to me.